# Generated by Django 5.0.6 on 2024-06-19 23:06

import django.contrib.auth.password_validation
import django.contrib.auth.validators
import django.core.validators
import user.models
from django.db import migrations, models


class Migration(migrations.Migration):

    initial = True

    dependencies = [
    ]

    operations = [
        migrations.CreateModel(
            name='Customer',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('created_at', models.DateTimeField(auto_now_add=True)),
                ('updated_at', models.DateTimeField(auto_now=True)),
                ('username', models.CharField(error_messages={'unique': 'A user with that username already exists.'}, help_text='Required. 150 characters or fewer. Letters, digits and @/./+/-/_ only.', max_length=150, unique=True, validators=[django.contrib.auth.validators.UnicodeUsernameValidator()], verbose_name='username')),
                ('password', models.CharField(max_length=128, validators=[django.contrib.auth.password_validation.validate_password])),
                ('first_name', models.CharField(blank=True, max_length=150)),
                ('last_name', models.CharField(blank=True, max_length=150)),
                ('photo', models.ImageField(blank=True, default='profile_photos/default_profile_photo.png', upload_to=user.models.Person.profile_image_upload_to)),
                ('phone', models.CharField(max_length=40, validators=[django.core.validators.RegexValidator(message='Invalid phone number!', regex='^(\\+98|0)?9\\d{9}$')])),
                ('email', models.EmailField(max_length=254)),
                ('address', models.TextField(blank=True, null=True)),
                ('is_active', models.BooleanField(default=True)),
                ('balance', models.IntegerField(default=0)),
            ],
            options={
                'abstract': False,
            },
        ),
        migrations.CreateModel(
            name='Staff',
            fields=[
                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
                ('created_at', models.DateTimeField(auto_now_add=True)),
                ('updated_at', models.DateTimeField(auto_now=True)),
                ('username', models.CharField(error_messages={'unique': 'A user with that username already exists.'}, help_text='Required. 150 characters or fewer. Letters, digits and @/./+/-/_ only.', max_length=150, unique=True, validators=[django.contrib.auth.validators.UnicodeUsernameValidator()], verbose_name='username')),
                ('password', models.CharField(max_length=128, validators=[django.contrib.auth.password_validation.validate_password])),
                ('first_name', models.CharField(blank=True, max_length=150)),
                ('last_name', models.CharField(blank=True, max_length=150)),
                ('photo', models.ImageField(blank=True, default='profile_photos/default_profile_photo.png', upload_to=user.models.Person.profile_image_upload_to)),
                ('phone', models.CharField(max_length=40, validators=[django.core.validators.RegexValidator(message='Invalid phone number!', regex='^(\\+98|0)?9\\d{9}$')])),
                ('email', models.EmailField(max_length=254)),
                ('address', models.TextField(blank=True, null=True)),
                ('is_active', models.BooleanField(default=True)),
                ('salary', models.IntegerField(default=0)),
                ('role', models.CharField(choices=[('manager', 'manager'), ('barista', 'barista'), ('waiter', 'waiter'), ('waitress', 'waitress')], default='barista', max_length=25)),
            ],
            options={
                'abstract': False,
            },
        ),
    ]
